extensions/frontend-tiles/src/GraphicsProvider/GraphicRepresentationProvider.ts

```diff
@@ -0,0 +1,255 @@
+      Logger.logError(loggerCategory, `Failed loading Graphics Data for Source ${args.dataSource.id}`);
+      break;
+    }
+
+    const foundSources = result.exports.filter((x) => x.request.exportType === args.dataSource.type && (args.includeIncomplete || x.status === GraphicRepresentationStatus.Complete));
+    for (const foundSource of foundSources) {
+      const graphicRepresentation = {
+        displayName: foundSource.displayName,
+        representationId: foundSource.id,
+        status: foundSource.status,
+        format: args.format,
+        url: foundSource._links.mesh.href,
+        dataSource: {
+          iTwinId: args.dataSource.iTwinId,
+          id: foundSource.request.iModelId,
+          versionId: foundSource.request.changesetId,
+          type: foundSource.request.exportType,
+        },
+      };
+
+      yield graphicRepresentation;
+    }
+
+    url = result._links.next?.href;
+  }
+}
+
+/** Arguments supplied  to [[obtainGraphicRepresentationUrl]].
+ * @beta
+ */
+export interface ObtainGraphicRepresentationUrlArgs {
+  /** The token used to access the mesh export service. */
+  accessToken: AccessToken;
+  /** The Session Id */
+  sessionId: string;
+  /** The data source for which to query */
+  dataSource: DataSource;
+  /** The format of the Graphics Data Source */
+  format: GraphicRepresentationFormat;
+  /** Chiefly used in testing environments. */
+  urlPrefix?: string;
+  /** If true, only Graphics Data produced for a specific version will be considered; otherwise, if no Graphics Data Sources are found for the version,
+  * the most recent source for any version will be used.
+   */
+  requireExactVersion?: boolean;
+  /** If true, enables a CDN (content delivery network) to access tiles faster. */
+  enableCDN?: boolean;
+}
+
+/** Obtains a URL pointing to a Graphic Representation.
+ * [[queryGraphicRepresentations]] is used to obtain a list of available representations. By default, the list is sorted from most to least recently-created.
+ * The first representation matching the source version is selected; or, if no such representation exists, the first representation in the list is selected.
+ * @returns A URL from which the tileset can be loaded, or `undefined` if no appropriate URL could be obtained.
+ * @beta
+ */
+export async function obtainGraphicRepresentationUrl(args: ObtainGraphicRepresentationUrlArgs): Promise<URL | undefined> {
+  if (!args.dataSource.id) {
+    Logger.logInfo(loggerCategory, "Cannot obtain Graphics Data from a source without an Id");
+    return undefined;
+  }
+
+  const queryArgs: QueryGraphicRepresentationsArgs = {
+    accessToken: args.accessToken,
+    sessionId: args.sessionId,
+    dataSource: {
+      iTwinId: args.dataSource.iTwinId,
+      id: args.dataSource.id,
+      changeId: args.dataSource.changeId,
+      type: args.dataSource.type,
+    },
+    format: args.format,
+    urlPrefix: args.urlPrefix,
+    enableCDN: args.enableCDN,
+  };
+
+  let selectedData;
```

Comment:
I'm curious if co-pilot could simplify this using "find". Do not change it this PR, it is just an idea to consider in the future.